kernel/src/engine/arrow_expression.rs

```diff
@@ -110,43 +111,77 @@ fn wrap_comparison_result(arr: BooleanArray) -> ArrayRef {
     Arc::new(arr) as Arc<dyn Array>
 }
 
-trait ProvidesColumnByName {
+trait ExtractableColumn {
```

Comment:
nit: why the name change? what exactly are you extracting here since you only being out a reference, I think the original name was more obvious albeit I may have named `ColumnByNameProvider` (enterprise java, sorry)

Reply:
reverted this